src/io/video.js

```diff
@@ -151,59 +113,11 @@ class Video {
         format = Video.FORMAT_IMAGE_DATA,
         cacheTimeout = this._frameCacheTimeout
     }) {
+        if (this.provider) return this.provider.getFrame({dimensions, mirror, format, cacheTimeout});
         if (!this.videoReady) {
```

Comment:
I don't think this block is needed here, provider does it. The early exit when not video ready will be handled above.